### PR TITLE
[Merged by Bors] - feat: the Fourier transform of a Schwartz function is differentiable

### PR DESCRIPTION
**Pull request by sgouezel**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Distribution/FourierSchwartz.lean [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  apply this.mono ((aestronglyMeasurable_id.norm.pow _).mul f.continuous.aestronglyMeasurable.norm)
+    (eventually_of_forall (fun x ↦ ?_))
+  conv_rhs => rw [norm_of_nonneg (by positivity), rpow_neg (by positivity), ← div_eq_mul_inv]
+  rw [le_div_iff' (by positivity)]
+  simp only [id_eq, Pi.mul_apply, Pi.pow_apply, norm_mul, norm_pow, norm_norm, rpow_nat_cast]
+  calc
+  (1 + ‖x‖) ^ (finrank ℝ D + 1) * (‖x‖ ^ k * ‖f x‖)
+    ≤ (1 + ‖x‖) ^ (finrank ℝ D + 1) * ((1 + ‖x‖) ^ k * ‖f x‖) := by gcongr; simp
+  _ = (1 + ‖x‖) ^ (finrank ℝ D + 1 + k) * ‖f x‖ := by simp [pow_add]; ring
+  _ ≤ C := hC x
```
Comment (mcdoll) [inline Mathlib/Analysis/Distribution/FourierSchwartz.lean:64]: I haven't seen `a \n < b`, only `a < \n b`, but I think that the later is bad, especially for inequalities.
```suggestion
  calc
    (1 + ‖x‖) ^ (finrank ℝ D + 1) * (‖x‖ ^ k * ‖f x‖)
      ≤ (1 + ‖x‖) ^ (finrank ℝ D + 1) * ((1 + ‖x‖) ^ k * ‖f x‖) := by gcongr; simp
    _ = (1 + ‖x‖) ^ (finrank ℝ D + 1 + k) * ‖f x‖ := by simp [pow_add]; ring
    _ ≤ C := hC x
```

File: Mathlib/Analysis/Distribution/FourierSchwartz.lean [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+lemma SchwartzMap.integrable [MeasurableSpace D] [BorelSpace D] [FiniteDimensional ℝ D]
+    (f : 𝓢(D, V)) {μ : Measure D} [IsAddHaarMeasure μ] :
+    Integrable f μ :=
+  (f.integrable_pow_mul (μ := μ) 0).mono f.continuous.aestronglyMeasurable
+    (eventually_of_forall (fun x ↦ by simp))
```
Comment (mcdoll) [inline Mathlib/Analysis/Distribution/FourierSchwartz.lean:70]:
```suggestion
  (f.integrable_pow_mul (μ := μ) 0).mono f.continuous.aestronglyMeasurable
    (eventually_of_forall (fun _ ↦ by simp))
```

File: Mathlib/Analysis/Distribution/SchwartzSpace.lean [hunk 3]
```diff
@@ -632,6 +631,36 @@ theorem _root_.Function.HasTemperateGrowth.norm_iteratedFDeriv_le_uniform_aux {f
...
+  refine ⟨contDiff_top_iff_fderiv.2 ⟨hf, h'f.1⟩ , fun n ↦ ?_⟩
+  rcases n with rfl|m
+  · exact ⟨k, C, fun x ↦ by simpa using h x⟩
+  · rcases h'f.2 m with ⟨k', C', h'⟩
+    refine ⟨k', C', fun x ↦ ?_⟩
+    simpa only [ContinuousLinearMap.strongUniformity_topology_eq, Function.comp_apply,
+      LinearIsometryEquiv.norm_map, iteratedFDeriv_succ_eq_comp_right] using h' x
+
+lemma _root_.Function.HasTemperateGrowth.zero :
+    Function.HasTemperateGrowth (fun _ : E ↦ (0 : F)) := by
```
Comment (mcdoll) [inline Mathlib/Analysis/Distribution/SchwartzSpace.lean:647]: I would assume that this is called `_root_.hasTemperateGrowth_zero`

File: Mathlib/Analysis/Distribution/FourierSchwartz.lean [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    exact Nat.cast_lt.mpr Nat.le.refl
+  apply this.mono ((aestronglyMeasurable_id.norm.pow _).mul f.continuous.aestronglyMeasurable.norm)
+    (eventually_of_forall (fun x ↦ ?_))
+  conv_rhs => rw [norm_of_nonneg (by positivity), rpow_neg (by positivity), ← div_eq_mul_inv]
+  rw [le_div_iff' (by positivity)]
+  simp only [id_eq, Pi.mul_apply, Pi.pow_apply, norm_mul, norm_pow, norm_norm, rpow_nat_cast]
+  calc
+  (1 + ‖x‖) ^ (finrank ℝ D + 1) * (‖x‖ ^ k * ‖f x‖)
+    ≤ (1 + ‖x‖) ^ (finrank ℝ D + 1) * ((1 + ‖x‖) ^ k * ‖f x‖) := by gcongr; simp
+  _ = (1 + ‖x‖) ^ (finrank ℝ D + 1 + k) * ‖f x‖ := by simp [pow_add]; ring
```
Comment (mcdoll) [inline Mathlib/Analysis/Distribution/FourierSchwartz.lean:63]: Does `simp only [pow_add]` also work?

**Review by mcdoll — COMMENTED** [review]
I have only very minor comments

File: Mathlib/Analysis/Distribution/FourierSchwartz.lean [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+Fourier transform of `VectorFourier.mul_L_schwartz L f`. -/
+def VectorFourier.mul_L_schwartz : 𝓢(D, V) →L[ℝ] 𝓢(D, E →L[ℝ] V) :=
+  -(2 * π * I) • (bilinLeftCLM (ContinuousLinearMap.smulRightL ℝ E V).flip L.hasTemperateGrowth)
+
+lemma VectorFourier.mul_L_schwartz_apply (f : 𝓢(D, V)) (d : D) :
+    VectorFourier.mul_L_schwartz L f d = VectorFourier.mul_L L f d := rfl
+
+lemma SchwartzMap.integrable_pow_mul [MeasurableSpace D] [BorelSpace D] [FiniteDimensional ℝ D]
+    (f : 𝓢(D, V)) {μ : Measure D} (k : ℕ) [IsAddHaarMeasure μ] :
+    Integrable (fun x ↦ ‖x‖^k * ‖f x‖) μ := by
```
Comment (mcdoll) [inline Mathlib/Analysis/Distribution/FourierSchwartz.lean:45]:
```suggestion
    Integrable (fun x ↦ ‖x‖ ^ k * ‖f x‖) μ := by
```

File: Mathlib/Analysis/Distribution/FourierSchwartz.lean (hunk 4 above)
Comment (sgouezel) [reply Mathlib/Analysis/Distribution/FourierSchwartz.lean]: No, but `simp only [pow_add, mul_assoc]` does. I've made the change.

**Comment by mcdoll** [comment]
maintainer merge

**Comment by ocfnash** [comment]
bors merge

**Pull request closed without merging**